src/components/Drawer/Drawer.tsx

```diff
@@ -0,0 +1,200 @@
+import clsx from "clsx";
+import {
+	type KeyboardEvent,
+	type KeyboardEventHandler,
+	useEffect,
+	useId,
+	useState,
+} from "react";
+
+import FocusLock from "react-focus-lock";
+import { handleAccessbilityError } from "utils";
+
+import "./Drawer_shim.css";
+import { IconButton } from "components/IconButton";
+import { Keys } from "utils";
+
+type EnforcedAccessibleLabel =
+	| {
+			title: string | JSX.Element;
+			actions?: JSX.Element[];
+			"aria-label"?: string;
+			"aria-labelledby"?: string;
+	  }
+	| {
+			title?: string | JSX.Element;
+			"aria-label": string;
+			"aria-labelledby"?: string;
+	  }
+	| {
+			title?: string | JSX.Element;
+			"aria-label"?: string;
+			"aria-labelledby": string;
+	  };
+
+interface BaseDrawerProps
+	extends Omit<React.HTMLAttributes<HTMLDialogElement>, "title"> {
+	children?: React.ReactNode;
+	id?: string;
+	onBack?: () => void;
+	onClose?: () => void;
+	closeOnBackgroundClick?: boolean;
+	open?: boolean;
+	width?: string;
+}
+
+export type DrawerProps = BaseDrawerProps & EnforcedAccessibleLabel;
+
+/**
+ * This component is used as a container of components that are dismisable.
+ *
+ * @example
+ * <Drawer open={isOpen} title="Mini Form">
+ *  <Form>
+ *   <TextInput />
+ *   <TextInput />
+ *  </Form>
+ * </Drawer>
+ *
+ * @see https://design.avayacloud.com/components/web/drawer-web
+ */
+export const Drawer = ({
+	children,
+	className,
+	id,
+	open = true,
+	title,
+	onBack,
+	onClose,
+	closeOnBackgroundClick = true,
+	width,
+
+	...rest
+}: DrawerProps) => {
+	const generatedId = useId();
+	id = id || generatedId;
```

Comment:
I think that you can update line 64 to be: `id = useId()`. But if you can't update these lines to:
```suggestion
	const generatedId = id || useId();
```

Reply:
done.